sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/rest/ReflectionSerializable.java

```diff
@@ -0,0 +1,338 @@
+                .getTarget()
+                .invoke();
+
+            handle = lookup.unreflect(xmlWriter.getDeclaredMethod("writeXml", xmlSerializable));
+            xmlWriterWriteXmlSerializable = (XmlWriterWriteXml) LambdaMetafactory.metafactory(defaultLookup, "writeXml",
+                    methodType(XmlWriterWriteXml.class), methodType(Object.class, Object.class, Object.class), handle,
+                    handle.type())
+                .getTarget()
+                .invoke();
+
+            xmlSerializableSupported = true;
+        } catch (Throwable e) {
+            if (e instanceof LinkageError || e instanceof Exception) {
+                LOGGER.log(LogLevel.VERBOSE, () -> "XmlSerializable serialization and deserialization isn't supported. "
+                    + "If it is required add a dependency of 'com.azure:azure-xml', or another dependencies which "
+                    + "include 'com.azure:azure-xml' as a transitive dependency. If your application runs as expected "
+                    + "this informational message can be ignored.");
+            } else {
+                throw (Error) e;
+            }
+        }
+
+        XML_SERIALIZABLE = xmlSerializable;
+        XML_READER = xmlReader;
+        XML_READER_CREATOR = xmlReaderCreator;
+        XML_WRITER_CREATOR = xmlWriterCreator;
+        XML_WRITER_WRITE_XML_START_DOCUMENT = xmlWriterWriteStartDocument;
+        XML_WRITER_WRITE_XML_SERIALIZABLE = xmlWriterWriteXmlSerializable;
+        XML_SERIALIZABLE_SUPPORTED = xmlSerializableSupported;
+        FROM_XML_CACHE = XML_SERIALIZABLE_SUPPORTED ? new ConcurrentHashMap<>() : null;
+    }
+
+    /**
+     * Whether {@code JsonSerializable} is supported and the {@code bodyContentClass} is an instance of it.
+     *
+     * @param bodyContentClass The body content class.
+     * @return Whether {@code bodyContentClass} can be used as {@code JsonSerializable}.
+     */
+    public static boolean supportsJsonSerializable(Class<?> bodyContentClass) {
+        return JSON_SERIALIZABLE_SUPPORTED && JSON_SERIALIZABLE.isAssignableFrom(bodyContentClass);
+    }
+
+    /**
+     * Serializes the {@code jsonSerializable} as an instance of {@code JsonSerializable}.
+     *
+     * @param jsonSerializable The {@code JsonSerializable} body content.
+     * @return The {@link ByteBuffer} representing the serialized {@code jsonSerializable}.
+     * @throws IOException If an error occurs during serialization.
+     */
+    static ByteBuffer serializeAsJsonSerializable(Object jsonSerializable) throws IOException {
+        AccessibleByteArrayOutputStream outputStream = new AccessibleByteArrayOutputStream();
+        try (Closeable jsonWriter = JSON_WRITER_CREATOR.createJsonWriter(outputStream)) {
+            JSON_WRITER_WRITE_JSON_SERIALIZABLE.writeJson(jsonWriter, jsonSerializable);
+        }
+
+        return ByteBuffer.wrap(outputStream.toByteArray(), 0, outputStream.count());
```

Comment:
should we close outputStream ?

Reply:
I can update it to do so, closing `AccessibleByteArrayOutputStream` is like closing `ByteArrayOutputStream` and is a no-op but if we ever change the implementation of `OutputStream` used this could hold on to resources